app/javascript/components/patient/vaccines/VaccineTable.js

```diff
@@ -0,0 +1,421 @@
+        </Dropdown.Menu>
+      </Dropdown>
+    );
+  };
+
+  /**
+   * Gets the data for the current vaccine if there is one selected/being edited.
+   */
+  getCurrVaccine = () => {
+    return this.state.editRow !== null && !!this.state.table.rowData ? this.state.table.rowData[this.state.editRow] : {};
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <Card id="vaccines-table" className="mx-2 mt-3 mb-4 card-square">
+          <Card.Header className="h5">Vaccinations</Card.Header>
+          <Card.Body>
+            <div className="mt-4">
+              <Row className="my-4">
+                <Col>
+                  <Button variant="primary" className="mr-2" onClick={this.handleAddVaccineClick}>
+                    <i className="fas fa-plus fa-fw"></i>
+                    <span className="ml-2">Add New Vaccine</span>
+                  </Button>
+                </Col>
+                <Col lg={5}>
+                  <InputGroup size="md">
+                    <InputGroup.Prepend>
+                      <OverlayTrigger overlay={<Tooltip>Search by ID, Group Name, or Product Name.</Tooltip>}>
+                        <InputGroup.Text className="rounded-0">
+                          <i className="fas fa-search"></i>
+                          <label htmlFor="vaccines-search-input" className="ml-1 mb-0">
+                            Search
+                          </label>
+                        </InputGroup.Text>
+                      </OverlayTrigger>
+                    </InputGroup.Prepend>
+                    <Form.Control
+                      id="vaccines-search-input"
+                      autoComplete="off"
+                      size="md"
+                      name="search"
+                      onChange={this.handleSearchChange}
+                      aria-label="Search"
+                    />
+                  </InputGroup>
+                </Col>
+              </Row>
+              <div className="mb-4">
+                <CustomTable
+                  columnData={this.state.table.colData}
+                  rowData={this.state.table.rowData}
+                  totalRows={this.state.table.totalRows}
+                  handleTableUpdate={query => this.updateTable({ ...this.state.query, order: query.orderBy, page: query.page, direction: query.sortDirection })}
+                  handleEntriesChange={this.handleEntriesChange}
+                  isLoading={this.state.isLoading}
+                  page={this.state.query.page}
+                  handlePageUpdate={this.handlePageUpdate}
+                  entryOptions={this.state.entryOptions}
+                  entries={this.state.query.entries}
+                />
+              </div>
+            </div>
+          </Card.Body>
+        </Card>
+        {(this.state.showAddVaccineModal || this.state.showEditVaccineModal) && (
+          <VaccineModal
+            title={this.state.showAddVaccineModal ? 'Add New Vaccination' : 'Edit Vaccination'}
```

Comment:
should there be consistent language between the 'Add New Vaccine' button and the 'Add New Vaccination' modal title?

Reply:
Yes, updated here: https://github.com/SaraAlert/SaraAlert/pull/721/commits/7b5ec1bd59c6c8c33fdb9331c5cf35933767c972#diff-36bdccbc821e45bb9795ad5d5f73dbf2f7030b1837d7590d14dccb54d8cd3eb1R350